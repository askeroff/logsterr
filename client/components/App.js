import React from 'react';
import Layout from './Layout';

const App = () => (
  <Layout>
    <h1 className="page-title">TimeTracker</h1>
  </Layout>
);

export default App;
