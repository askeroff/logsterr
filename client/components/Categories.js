import React from 'react';
import Layout from './layout/Layout';

const Categories = () => (
  <Layout>
    <h1 className="page-title">Categories</h1>
    <p>This page will list all the categories now available;</p>
    <p>as well as let the user add new ones </p>
  </Layout>
);

export default Categories;
