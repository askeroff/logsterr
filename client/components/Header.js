import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Logout from './Logout';
import '../styles/header.styl';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.guestLinks = this.guestLinks.bind(this);
  }

  guestLinks() {
    if (!this.props.user) {
      return (
        <ul className="header-menu">
          <li className="header-menu-item">
            <Link to="/">Home</Link>
          </li>
          <li className="header-menu-item">
            <Link to="/signup">Sign Up</Link>
          </li>
          <li className="header-menu-item">
            <Link to="/login">Log In</Link>
          </li>
        </ul>
      );
    }
    return (
      <ul className="header-menu">
        <li className="header-menu-item">
          <Link to="/">Home</Link>
        </li>
        <Logout history={this.props.history} />
      </ul>
    );
  }

  render() {
    return (
      <header>
        { this.guestLinks() }
      </header>
    );
  }
}

Header.defaultProps = {
  user: null,
  history: {},
};

Header.propTypes = {
  user: PropTypes.object,
  history: PropTypes.object,
};

export default Header;
