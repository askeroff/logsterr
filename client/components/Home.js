import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Layout from './layout/Layout';
import Dashboard from './dashboard/Index';
import Spinner from './layout/Spinner';

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loggedIn: false,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.user.loggedIn === true) {
      this.setState({ loggedIn: true });
    } else if (nextProps.user.loggedIn === false) {
      this.props.history.push('/login');
    }
  }

  shouldRender() {
    if (this.state.loggedIn) {
      return <Dashboard />;
    }
    return <Spinner />;
  }

  render() {
    return (
      <Layout>
        {this.shouldRender()}
      </Layout>
    );
  }
}

Home.propTypes = {
  user: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  user: state.user,
});

export default connect(mapStateToProps)(Home);
