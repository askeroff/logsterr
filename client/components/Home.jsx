"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var React = require("react");
var react_redux_1 = require("react-redux");
var Layout_1 = require("./layout/Layout");
var Index_1 = require("./dashboard/Index");
var Spinner_1 = require("./layout/Spinner");
var LandingPage_1 = require("./LandingPage");
var Home = /** @class */ (function (_super) {
    __extends(Home, _super);
    function Home() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.shouldRender = function () {
            var hasHistory = Object.prototype.hasOwnProperty.call(_this.props.history, 'push');
            if (_this.props.user.loggedIn) {
                return (<Layout_1.default showToGuests>
          <Index_1.default />
        </Layout_1.default>);
            }
            else if (_this.props.user.loggedIn === false && hasHistory) {
                return (<Layout_1.default showToGuests contentClass="landing-wrapper">
          <LandingPage_1.default />
        </Layout_1.default>);
            }
            return (<Layout_1.default showToGuests>
        <Spinner_1.default />
      </Layout_1.default>);
        };
        return _this;
    }
    Home.prototype.render = function () {
        return this.shouldRender();
    };
    Home.defaultProps = { history: {}, user: {} };
    return Home;
}(React.Component));
var mapStateToProps = function (state) { return ({
    user: state.user
}); };
exports["default"] = react_redux_1.connect(mapStateToProps)(Home);
