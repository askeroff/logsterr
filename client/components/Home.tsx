import * as React from 'react';
import { connect } from 'react-redux';
import Layout from './layout/Layout';
import Dashboard from './dashboard/Index';
import Spinner from './layout/Spinner';
import { IUser } from '../types';
import LandingPage from './LandingPage';

interface Props {
  user: IUser,
  history: any
}

class Home extends React.Component<Props, any> {
  static defaultProps = { history: {}, user: {} };

  shouldRender = () => {
    const hasHistory = Object.prototype.hasOwnProperty.call(
      this.props.history,
      'push'
    );
    if (this.props.user.loggedIn) {
      return (
        <Layout showToGuests>
          <Dashboard />
        </Layout>
      );
    } else if (this.props.user.loggedIn === false && hasHistory) {
      return (
        <Layout showToGuests contentClass="landing-wrapper">
          <LandingPage />
        </Layout>
      );
    }
    return (
      <Layout showToGuests>
        <Spinner />
      </Layout>
    );
  };

  render() {
    return this.shouldRender();
  }
}

const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps)(Home);
