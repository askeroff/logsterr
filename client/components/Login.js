import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { logIn } from '../actions/user';
import LoginForm from './layout/LoginForm';
import Layout from './layout/Layout';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
    };
    this.handleEmailChange = this.handleEmailChange.bind(this);
    this.handlePasswordChange = this.handlePasswordChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.user.email !== undefined) {
      this.props.history.push('/');
    }
  }

  handleEmailChange(event) {
    this.setState({ email: event.target.value });
  }

  handlePasswordChange(event) {
    this.setState({ password: event.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.handleLogin({
      email: this.state.email,
      password: this.state.password,
    });
  }

  render() {
    return (
      <Layout>
        <h1 className="page-title">Login Page</h1>
        <LoginForm
          myClassName="form signup"
          handleSubmit={this.handleSubmit}
          emailValue={this.state.email}
          passwordValue={this.state.password}
          handleEmailChange={this.handleEmailChange}
          handlePasswordChange={this.handlePasswordChange}
        />
        <div
          ref={resultMessage => {
            this.resultMessage = resultMessage;
          }}
        />
      </Layout>
    );
  }
}

Login.propTypes = {
  handleLogin: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  user: state.user,
});

const mapDispatchToProps = dispatch => ({
  handleLogin(user) {
    dispatch(logIn(user));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
