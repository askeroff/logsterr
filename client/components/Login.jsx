// @flow
import React from 'react';
import { connect } from 'react-redux';
import { logIn, forgot } from '../actions/user';
import LoginForm from './layout/LoginForm';
import ForgotForm from './layout/ForgotForm';
import Layout from './layout/Layout';
import Spinner from './layout/Spinner';
import { IUser } from '../types';

type Props = {
  user: IUser,
  history: any,
  error: string,
  handleLogin: ({ email: string, password: string }) => void,
  handleForgot: (email: string) => void
};

type State = {
  email: string,
  forgotEmail: string,
  password: string,
  spinner: boolean,
  forgotSpinner: boolean
};

class Login extends React.Component<Props, State> {
  static defaultProps = {
    error: ''
  };
  state = {
    email: '',
    forgotEmail: '',
    password: '',
    spinner: false,
    forgotSpinner: false
  };

  componentDidMount() {
    this.props.user.error = '';
  }

  componentWillReceiveProps(nextProps: any) {
    if (nextProps.user.forgotResponse !== undefined) {
      this.setState({ forgotSpinner: false });
    }
    if (nextProps.user.email !== undefined) {
      this.props.history.push('/');
    }

    if (nextProps.error !== '' && nextProps.error !== undefined) {
      this.setState({ spinner: false });
      if (this.resultMessage !== null) {
        this.resultMessage.innerHTML = nextProps.error;
      }
    }
  }

  resultMessage: HTMLDivElement | null;

  handleEmailChange = (event: SyntheticEvent<HTMLInputElement>) => {
    this.setState({ email: event.currentTarget.value });
  };

  handleForgotEmailChange = (event: SyntheticEvent<HTMLInputElement>) => {
    this.setState({ forgotEmail: event.currentTarget.value });
  };

  handlePasswordChange = (event: SyntheticEvent<HTMLInputElement>) => {
    this.setState({ password: event.currentTarget.value });
  };

  handleSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
    event.preventDefault();
    this.setState({ spinner: true });
    if (this.resultMessage !== null) {
      this.resultMessage.innerHTML = '';
    }
    this.props.handleLogin({
      email: this.state.email,
      password: this.state.password
    });
  };

  handleForgotSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
    event.preventDefault();
    this.props.handleForgot(this.state.forgotEmail);
    this.setState({ forgotSpinner: true });
  };

  render() {
    return (
      <Layout>
        <h1 className="page-title">Login Page</h1>
        <LoginForm
          myClassName="form__login"
          handleSubmit={this.handleSubmit}
          emailValue={this.state.email}
          passwordValue={this.state.password}
          handleEmailChange={this.handleEmailChange}
          handlePasswordChange={this.handlePasswordChange}
        />
        <div
          className="server-response"
          ref={resultMessage => {
            this.resultMessage = resultMessage;
          }}
        />
        {this.state.spinner ? <Spinner /> : null}
        <h2 className="page-title">Did you forget your password?</h2>
        <ForgotForm
          myClassName="form__login"
          emailValue={this.state.forgotEmail}
          handleEmailChange={this.handleForgotEmailChange}
          handleSubmit={this.handleForgotSubmit}
        />
        {this.state.forgotSpinner ? <Spinner /> : null}
      </Layout>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user,
  error: state.user.error
});

const mapDispatchToProps = dispatch => ({
  handleLogin(user) {
    dispatch(logIn(user));
  },
  handleForgot(email) {
    dispatch(forgot(email));
  }
});

export const UnwrappedLogin = Login;
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);
