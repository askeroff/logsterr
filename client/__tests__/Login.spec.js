import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { UnwrappedLogin } from '../components/Login';

Enzyme.configure({ adapter: new Adapter() });

test('Login Component renders correctly', () => {
  const component = shallow(
    <UnwrappedLogin
      handleLogin={() => 1}
      user={{ email: 'test@gmail.com' }}
      history={{}}
    />
  );
  expect(component).toMatchSnapshot();
});
