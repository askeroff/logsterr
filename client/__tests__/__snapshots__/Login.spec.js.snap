// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<Connect(Layout)>
  <h1
    className="page-title"
  >
    Login Page
  </h1>
  <LoginForm
    emailValue=""
    handleEmailChange={[Function]}
    handlePasswordChange={[Function]}
    handleSubmit={[Function]}
    myClassName="form signup"
    passwordValue=""
  />
  <div
    className="server-response"
  />
</Connect(Layout)>
`;
